File: 1021 - Remove Outermost Parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        String res = "";
        int d = 0;
        
        for (char c : s.toCharArray()) {
            if (c == ')') {
                d--;
            } 
            if (d > 0) res += c;
            if (c == '(') {
                d++;
            }
        }
        return res;
    }
}

File: 1021 - Remove Outermost Parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function (s) {
    let res = '', d = 0;

    for (let c of s) {
        if (c == ')') d--;
        if (d > 0) res += c;
        if (c == '(') d++;
    }
    return res;
};